text/0192-remove-constructs-compat.md

```diff
@@ -0,0 +1,888 @@
+API shim so that CDK code will continue to function without change.
+
+As part [AWS CDK v2.0], we plan to remove this redundant layer and make some
+improvements to a few APIs in "constructs" based on our learnings from CDK v1.x
+and new projects such as CDK for Kubernetes.
+
+This RFC describes the motivation, implications and plan for this project.
+
+[AWS CDK v2.0]: https://github.com/aws/aws-cdk-rfcs/issues/156
+[construct-compat.ts]: https://github.com/aws/aws-cdk/blob/fcca86e82235387b88371a0682cd0fc88bc1b67e/packages/%40aws-cdk/core/lib/construct-compat.ts
+
+# Table of Contents
+
+- [Summary](#summary)
+- [Table of Contents](#table-of-contents)
+- [Release Notes](#release-notes)
+  - [01-BASE-TYPES: Removal of base types](#01-base-types-removal-of-base-types)
+  - [02-ASPECTS: Changes in Aspects API](#02-aspects-changes-in-aspects-api)
+  - [03-DEPENDABLE: Changes to IDependable implementation](#03-dependable-changes-to-idependable-implementation)
+  - [04-STACK-ROOT: Stacks as root constructs](#04-stack-root-stacks-as-root-constructs)
+  - [05-METADATA-TRACES: Stack traces no longer attached to metadata by default](#05-metadata-traces-stack-traces-no-longer-attached-to-metadata-by-default)
+  - [06-NO-PREPARE: The `prepare` hook is no longer supported](#06-no-prepare-the-prepare-hook-is-no-longer-supported)
+  - [07-NO-SYNTHESIZE: The `synthesize` hook is no longer supported](#07-no-synthesize-the-synthesize-hook-is-no-longer-supported)
+  - [08-VALIDATION: The `validate()` hook is now `node.addValidation()`](#08-validation-the-validate-hook-is-now-nodeaddvalidation)
+  - [09-LOGGING: Logging API changes](#09-logging-logging-api-changes)
+- [Motivation](#motivation)
+  - [1. Redundant layer](#1-redundant-layer)
+  - [2. Multiple `Construct` types](#2-multiple-construct-types)
+  - [3. Composability with other domains](#3-composability-with-other-domains)
+  - [4. Non-intuitive dependency requirement](#4-non-intuitive-dependency-requirement)
+- [Design](#design)
+  - [01-BASE-TYPES](#01-base-types)
+  - [02-ASPECTS](#02-aspects)
+  - [03-DEPENDABLE](#03-dependable)
+  - [04-STACK-ROOT](#04-stack-root)
+  - [05-METADATA-TRACES](#05-metadata-traces)
+  - [06-NO-PREPARE](#06-no-prepare)
+  - [07-NO-SYNTHESIZE](#07-no-synthesize)
+  - [08-VALIDATION](#08-validation)
+  - [09-LOGGING](#09-logging)
+- [Drawbacks](#drawbacks)
+  - [User migration effort](#user-migration-effort)
+  - [CDK codebase migration efforts](#cdk-codebase-migration-efforts)
+  - [Staging of the 2.x fork](#staging-of-the-2x-fork)
+- [Rationale and Alternatives](#rationale-and-alternatives)
+  - [Alternatives considered](#alternatives-considered)
+- [Adoption Strategy](#adoption-strategy)
+- [Unresolved questions](#unresolved-questions)
+- [Future Possibilities](#future-possibilities)
+- [Implementation Plan](#implementation-plan)
+  - [Preparation of 1.x](#preparation-of-1x)
+  - [constructs 4.x](#constructs-4x)
+  - [2.x Work](#2x-work)
+
+# Release Notes
```

Comment:
What about the developer guide, and our documentation.

We should include a section in this RFC (or maybe in the one for v2) on how that is going to look when both v1 and v2 are present, just to make sure we don't forget about it.

Reply:
added tasks